rust/parser/src/lexer.rs

```diff
@@ -344,11 +344,11 @@ impl<'alloc> Lexer<'alloc> {
         // SourceCharacter elements. A code point in a keyword cannot be
         // expressed by a `\` UnicodeEscapeSequence.
         let id = if has_different {
-            // Always return `Name`.
+            // Always return `NameMaybeKeyword`.
             //
             // Error check against reserved word should be handled in the
             // consumer.
-            TerminalId::Name
+            TerminalId::NameMaybeKeyword
```

Comment:
To me, `NameMaybeKeyword` sounds like "input that matches *IdentifierName* but might also be a keyword". But it actually means "input that matches *IdentifierName* and might have the same StringValue as a keyword, but definitely **can't** be a keyword, because it contains escapes".

Consider `TerminalId::NameWithEscape` or some other name.

Reply:
Changed to TerminalId::NameWithEscape